autoPyTorch/pipeline/components/training/trainer/GridCutOutTrainer.py

```diff
@@ -0,0 +1,56 @@
+import typing
+
+import numpy as np
+
+from autoPyTorch.pipeline.components.training.trainer.base_trainer import BaseTrainerComponent
+from autoPyTorch.pipeline.components.training.trainer.cutout_utils import CutOut
+
+
+class GridCutOutTrainer(CutOut, BaseTrainerComponent):
+
+    def data_preparation(self, X: np.ndarray, y: np.ndarray,
+                         ) -> typing.Tuple[np.ndarray, typing.Dict[str, np.ndarray]]:
+        """
+        Depending on the trainer choice, data fed to the network might be pre-processed
+        on a different way. That is, in standard training we provide the data to the
+        network as we receive it to the loader. Some regularization techniques, like mixup
+        alter the data.
+
+        Args:
+            X (np.ndarray): The batch training features
+            y (np.ndarray): The batch training labels
+
+        Returns:
+            np.ndarray: that processes data
+            typing.Dict[str, np.ndarray]: arguments to the criterion function
+        """
+        r = self.random_state.rand(1)
+        batch_size, channel, W, H = X.size()
+        if r > self.cutout_prob:
+            return X, {'y_a': y, 'y_b': y, 'lam': 1}
+
+        # Draw parameters of a random bounding box
+        # Where to cut basically
+        cut_rat = np.sqrt(1. - self.patch_ratio)
+        cut_w = np.int(W * cut_rat)
+        cut_h = np.int(H * cut_rat)
+        cx = self.random_state.randint(W)
+        cy = self.random_state.randint(H)
```

Comment:
I will also continue here by the example of `patch_size = 0.16` and `image_size = (5, 5)`.
So in this case, `cut_rat = np.sqrt(1. - 0.16) = 0.916515138991168` and `cut_w = cut_h = 4`.
It means we will get the following images:
```
(1) cx = 0, cy = 2
xxx##
xxx##
#####
#####
#####

(2) cx = 1, cy = 2
xxx##
xxx##
xxx##
#####
#####

(3) cx = 2, cy = 2
xxx##
xxx##
xxx##
xxx##
#####

(4) cx = 3, cy = 2
#####
xxx##
xxx##
xxx##
xxx##
```

I do not know if this procedure is really what you want to apply.